app/services/foreman/foreman_url_renderer.rb

```diff
@@ -39,20 +36,44 @@ def foreman_url(action, params = {}, unescaped_params = {})
         raw_string += unescaped_params.map { |k, v| "#{k}=#{v}" }.join('&')
       end
 
+      url_options[:action] = action
+      url_options[:path] = config.path
+      render_foreman_url(host, url_options, params) + raw_string
+    end
+
+    apipie :method, 'Returns Foreman request address' do
+      desc 'Returns FQDN + Port to Foreman or Smart Proxy depending on host.subnet.template proxy configuration.'
+      returns String, desc: "Rendered request address"
+    end
+    def foreman_request_addr
+      url_options = foreman_url_options[:url_options]
+
+      result = url_options[:host]
+      result += ":" + url_options[:port] unless url_options[:port].to_s.empty? || url_options[:port] == 80
```

Comment:
I guess it will be safer to use interpolation:
```suggestion
      result += ":#{url_options[:port]}" unless url_options[:port].to_s.empty? || url_options[:port] == 80
```
Otherwise it could yell `no implicit conversion of Integer into String`